### PR TITLE
fix potential memory management issues in cython code

### PR DESCRIPTION
**Pull request by noskill**
as discussed in pr https://github.com/opencog/atomspace/pull/2102

### CODE REVIEW

**Comment by noskill** [comment]
I need some time to figure out how to run circleci locally

**Comment by linas** [comment]
I think this is obsoleted by #2106?

**Comment by noskill** [comment]
it's not obsolete, changes are different. I was able to run tests in docker after some changes to circleci config - https://gist.github.com/noskill/9dc96b40f0c9b47acbb2110ac6ae95d5 but test passed. Maybe this pr will pass too.

**Comment by noskill** [comment]
Tests passed

File: opencog/cython/opencog/backwardchainer.pyx [hunk 1]
```diff
@@ -52,8 +52,7 @@ cdef class BackwardChainer:
         return result
 
     def __dealloc__(self):
-        self._control_as = None
+        del self.chainer
         self._trace_as = None
+        self._control_as = None
         self._as = None
```
Comment (vsbogd) [inline opencog/cython/opencog/backwardchainer.pyx]: Minor, but it seems logical to set `self.parent_atomspace = None` as well

Comment (noskill) [reply opencog/cython/opencog/backwardchainer.pyx]: fixed

**Comment by noskill** [comment]
opencog failed to build

**Pull request merged**